src/saving/NewSaveMenu.cs

```diff
@@ -132,10 +137,29 @@ private void OnSelectedChanged()
         usingSelectedSaveName = true;
     }
 
-    private void OnSaveNameTextEntered(string newName)
+    private void OnSaveNameTextChanged(string newName)
     {
-        _ = newName;
+        if (!string.IsNullOrWhiteSpace(newName) && !newName.Any(Constants.SAVE_NAME_BLACKLIST.Contains))
```

Comment:
I'd make this into a one liner helper method as this is a pretty long condition that needs to be kept in sync in two different methods, so having one helper method will guard against someone accidentally editing one place and not the other.